Replace Tango icons with Jenkins Symbols

janfaracik opened the pull request:
**Dashboard:**
<img width="331" alt="image" src="https://user-images.githubusercontent.com/43062514/157138346-d6797460-477d-46cc-abab-25405e66ef5f.png">
**Project:**
<img width="349" alt="image" src="https://user-images.githubusercontent.com/43062514/157138320-2b8976c9-057f-4641-838d-2c0495a43e40.png">
**Build:**
<img width="342" alt="image" src="https://user-images.githubusercontent.com/43062514/157138373-58468eca-2386-423b-9cd0-f5772ac98f52.png">

This is the 'biggie' of several PRs to update Jenkins' icons https://github.com/jenkinsci/jenkins/pull/6186, this PR:

* Updates the rest of Tango icons found in Jenkins to use [Jenkins Symbols](https://www.jenkins.io/doc/developer/views/symbols/)
* The sidebar menu design has been updated 
* Adds more symbols to the Tango-to-Symbol translation method
* Very slight tweak to the Manage Jenkins options corner radius

The majority of files changed in this PR are:

* Adding symbols
* Updating classes/pages to use Jenkins Symbols

---

I've added a file `buttons-temp.less` which adds the button styling from #5897 so that new buttons in this PR can take advantage of the new stylings without that PR being merged in yet - happy to remove it though if it's deemed out of scope.

---

Would be very interested to hear feedback on this PR + identify spots I've missed.

The follow up PR to this (or at least one very soon after this if issues pop up) will be to allow for custom symbols so that plugin developers can add their own symbols to use.

### Proposed changelog entries

* Update icons

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

<!-- For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc) -->

### Desired reviewers

@mention

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/code-reviewers
-->

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

janfaracik committed:
Init

janfaracik committed:
Update padding

janfaracik committed:
Init

janfaracik committed:
Update Icon.java

janfaracik committed:
Renamed SVGs folder from ionicons, add test icon

janfaracik committed:
Updated plugin manager

janfaracik committed:
Getting there

janfaracik committed:
Working (albeit very messy) build

janfaracik committed:
Working build

janfaracik committed:
Update Icon.java

janfaracik committed:
Rename item

janfaracik committed:
Rename param to symbol

janfaracik committed:
Update icon.jelly

janfaracik committed:
Working build

janfaracik committed:
Move functions to Java

janfaracik committed:
Cleanup + fix menu

janfaracik committed:
Fix settings symbol size

janfaracik committed:
Reset some files

janfaracik committed:
Address some comments

janfaracik committed:
Fix test + add case for symbol

janfaracik committed:
Add translation service

janfaracik committed:
Remove suppresion

janfaracik committed:
Icons

janfaracik committed:
Use symbols for manage jenkins

janfaracik committed:
Update icons + manage UI

janfaracik committed:
Update manage.jelly

janfaracik committed:
Update section.less

janfaracik committed:
Merge branch 'scalable-icon-support' into new-task-design

janfaracik committed:
Update side-panel-tasks.less

janfaracik committed:
Remove title and text decoration

janfaracik committed:
Translate more icons

janfaracik committed:
Create windows.svg

janfaracik committed:
Update section.less

janfaracik committed:
Update section.less

janfaracik committed:
Update section.less

janfaracik committed:
Add tooltip to update badge

janfaracik committed:
Add suggestions by @timja

janfaracik committed:
Fix breadcrumbs icon missing/incorrect size

janfaracik committed:
Potentially fix test

janfaracik committed:
Update Functions.java

janfaracik committed:
Potential fix for plugins icons not showing

janfaracik committed:
Fix missing slash in tests

janfaracik committed:
Restore icon requirement for tasks

janfaracik committed:
Merge branch 'scalable-icon-support' into scalable-icon-support-2-manage-jenkins

janfaracik committed:
Icons are now blue in dropdown menus, icon color no longer changes on manage page

janfaracik committed:
Merge branch 'scalable-icon-support' into scalable-icon-support-2-new-tasks

janfaracik committed:
Replace more icons

janfaracik committed:
Add icons for nodes

janfaracik committed:
Merge branch 'master' into scalable-icon-support

janfaracik committed:
Update fingerprint icon + page

janfaracik committed:
Update projectRelationship.jelly

janfaracik committed:
Merge branch 'scalable-icon-support' into scalable-icon-support-2-sidebar

janfaracik committed:
Add buttons temp

janfaracik committed:
Merge branch 'master' into scalable-icon-support

janfaracik committed:
Merge branch 'master' into scalable-icon-support

janfaracik committed:
Update repeatableDeleteButton.jelly

janfaracik committed:
Update Action javadoc

janfaracik committed:
Update summary

janfaracik committed:
Remove icon loading logic from summary

janfaracik committed:
Merge branch 'scalable-icon-support' into scalable-icon-support-2-manage-jenkins

janfaracik committed:
Update HudsonPrivateSecurityRealm.java

janfaracik committed:
Update jenkins.svg

janfaracik committed:
Update arrow-up.svg

janfaracik committed:
Update index.groovy

janfaracik committed:
Merge branch 'master' into scalable-icon-support-2-manage-jenkins

janfaracik committed:
Update plugins.svg

janfaracik committed:
Fix test

janfaracik committed:
Update section.less

janfaracik committed:
Revert "Update section.less"

This reverts commit 96a036f1594c86ae7eea03c781404227fe35a95a.

janfaracik committed:
Fix potential icon stretching on manage page

janfaracik committed:
Shrink table header slightly

janfaracik committed:
Add additional icons for nodes + users, set default user icon to person-circle

janfaracik committed:
Fix test + linting

janfaracik committed:
Fix oversized user icon

janfaracik committed:
Merge branch 'scalable-icon-support-2-manage-jenkins' into scalable-icon-support-2-sidebar

janfaracik committed:
fix

janfaracik committed:
Update SVG icon, remove duplicated new icon

janfaracik committed:
Update icon, fix sizes

janfaracik committed:
Add label icons

janfaracik committed:
Add more icons

janfaracik committed:
Add more icons

janfaracik committed:
Update sidebar tasks to be smaller

janfaracik committed:
Bold sidebar icons on selection

janfaracik committed:
Merge branch 'master' into scalable-icon-support-2-sidebar

janfaracik committed:
Fix spacing of manage jenkins page

janfaracik committed:
Merge branch 'master' into scalable-icon-support-2-sidebar

NotMyFault reviewed (changes requested):
Nice to see that more icons are migrating to a modern icon base.
Though I have to admit, that I actually liked the blue color of sidebar icons on the currently weekly 👀

A few things I noticed:

- Either are external icon sources displayed much larger than ionicons, or the credentials icon is biting back again
<details>
<summary></summary>

![](https://i.imgur.com/N0LfK4n.png)

</details>

- The workspace's alignment of icons, text and the searchbox is a bit out of order
<details>
<summary></summary>

![](https://i.imgur.com/EEVGitW.png)

</details>

- I'd vouch against using the list symbol as "Changes" icon, because this typical icon is often used as "menu" or slide icon on other applications and looks a bit misleading.

uhafner reviewed (commented):
The screenshots look awesome! 

I'm going on holidays for a week so I can't look into it in more detail now. But I think there are enough other contributors around that can look into it.

daniel-beck commented on core/src/main/resources/hudson/logging/LogRecorderManager/levels.jelly:
```diff
@@ -34,7 +34,7 @@ THE SOFTWARE.
       ${%Logger Configuration}
       <t:help iconSize="large" tooltip="${%Additional information on log levels}" href="${%url}" />
     </h1>
-    <table class="sortable pane jenkins-table" id="logLevels">
```
Unrelated?

timja replied:
It is related, without this change the help icon is not rendered correctly.

daniel-beck commented on core/src/main/resources/hudson/diagnosis/MemoryUsageMonitor/index.jelly:
```diff
@@ -29,7 +29,6 @@ THE SOFTWARE.
     <!-- TODO: st:include page="sidepanel.jelly" /-->
     <l:main-panel>
       <h1>
-        <l:icon class="icon-monitor icon-xlg"/>
```
Unrelated?

daniel-beck commented on core/src/main/resources/hudson/model/View/sidepanel.jelly:
```diff
@@ -51,7 +51,7 @@ THE SOFTWARE.
         </j:otherwise>
       </j:choose>
-      <l:task href="${rootURL}/${it.viewUrl}builds" icon="icon-notepad icon-md" title="${%Build History}"/>
+      <l:task href="${rootURL}/${it.viewUrl}builds" icon="symbol-build-history" title="${%Build History}"/>
```
This invokes the "Time Machine" UI or what is this icon?

janfaracik replied:
Aha it definitely does,. What do you think of this one/do you have a better icon in mind?

<img width="304" alt="image" src="https://user-images.githubusercontent.com/43062514/156634735-93425aff-9316-4d9a-af45-ba86909842a2.png">

daniel-beck replied:
That has more of a "record" feel, so probably? Time Machine seems too clever, and "going back in time" is probably more for https://plugins.jenkins.io/jobConfigHistory/ than build history.

daniel-beck commented on core/src/main/resources/jenkins/model/Jenkins/fingerprintCheck.jelly:
```diff
@@ -31,8 +31,7 @@ THE SOFTWARE.
       <h1>
-        <l:icon class="icon-fingerprint icon-xlg"/>
-        ${%Check File Fingerprint}
+        ${title}
```
Seems brittle to reuse the `title` attribute from `l:layout` here.

daniel-beck commented on core/src/main/resources/jenkins/model/Jenkins/projectRelationship.jelly:
```diff
@@ -28,9 +28,15 @@ THE SOFTWARE.
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
   <l:layout title="${%Project Relationship}">
+    <l:app-bar title="${title}">
```
Seems brittle to reuse the `title` attribute from `l:layout` here.

daniel-beck commented on core/src/main/resources/jenkins/model/Jenkins/projectRelationship.jelly:
```diff
@@ -28,9 +28,15 @@ THE SOFTWARE.
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
   <l:layout title="${%Project Relationship}">
+    <l:app-bar title="${title}">
+      <a class="jenkins-button" tooltip="${%Help}" href="projectRelationship-help">
+        <l:icon src="symbol-help-circle" class="icon-sm" />
+      </a>
+    </l:app-bar>
```
Unnecessary unrelated structural change? Why not fix up the icon inside `h1` and done?

daniel-beck commented on core/src/main/resources/hudson/logging/LogRecorderManager/sidepanel.jelly:
```diff
@@ -34,10 +34,10 @@ THE SOFTWARE.
         </l:isAdmin>
       </l:task>
-      <l:task href="all" icon="icon-notepad icon-md" title="${%All Log Messages}"/>
+      <l:task href="all" icon="icon-clipboard icon-md" title="${%All Log Messages}"/>
```
This used to be different from the main entry (line 35). How about using the same icon as used on Manage Jenkins for System Log here on line 35?

daniel-beck commented on core/src/main/resources/hudson/model/CauseAction/summary.jelly:
```diff
@@ -24,7 +24,7 @@ THE SOFTWARE.
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
-  <t:summary icon="orange-square.png">
+  <t:summary icon="symbol-person-circle">
```
Builds are not necessarily triggered by people, so this is a weird choice.

daniel-beck commented on core/src/main/resources/hudson/model/Label/index.jelly:
```diff
@@ -46,9 +45,7 @@ THE SOFTWARE.
-            <span class="icon-sm">
-              <a href="${url}"><l:icon src="${c.iconClassName}"/></a>
-            </span>
+            <a href="${url}" class="jenkins-link--with-icon"><l:icon src="${c.iconClassName}" /></a>
```
Why is this not a reusable Jelly tag?

daniel-beck commented on core/src/main/resources/hudson/model/Label/sidepanel.jelly:
```diff
@@ -32,11 +32,11 @@ THE SOFTWARE.
         <l:task href="${url}/configure" icon="symbol-settings" title="${%Configure}" permission="${app.ADMINISTER}" />
       </j:if>
-      <l:task href="${url}/load-statistics" icon="icon-monitor icon-md" title="${%Load Statistics}"/>
+      <l:task href="${url}/load-statistics" icon="symbol-terminal" title="${%Load Statistics}"/>
```
```suggestion
      <l:task href="${url}/load-statistics" icon="symbol-analytics" title="${%Load Statistics}"/>
```

Consistent with the others.

daniel-beck commented on core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly:
```diff
@@ -47,11 +47,15 @@ THE SOFTWARE.
     -->
     <span class="build-rss-links">
         <a class="build-rss-all-link" href="${it.baseUrl}/rssAll">
```
Why is this not a `jenkins-link--with-icon`?

daniel-beck commented on war/src/main/less/base/layout-commons.less:
```diff
@@ -69,10 +69,7 @@ body {
 }
 
 #side-panel {
-  padding: 1rem 0 2.5rem 0;
-  width: 300px;
   flex-shrink: 0;
-  border-right: 1px solid var(--panel-border-color);
```
Unrelated?

NotMyFault replied:
I wouldn't call it necessarily unrelated. That one removes the small gray delimiter that splits the sidepanel from the mainpanel. Appears to be a seamless transition integrating well with the new sidebar.

daniel-beck replied:
If your PR changes what icons look like, a change of border color/existence is unrelated.

These PRs are difficult enough to review without adding changes to entirely different UI elements.

daniel-beck reviewed (commented):
Plugin Manager and Update Center have the same icon (a puzzle piece). Is there no "Download" looking icon (e.g. a bold downward arrow outline) available for the latter? (Seems not introduced here.)

In case the following are unintentionally unchanged:
* The "right arrow" icons on the "Update Center" page related to Jenkins restart are unchanged and look out of place.
* <img width="548" alt="Screenshot 2022-02-24 at 20 39 40" src="https://user-images.githubusercontent.com/1831569/155595628-682789d0-bc0f-4f9f-b077-300abc10b2bb.png">

janfaracik committed:
Update status/changes icons, fix large workspace icon

NotMyFault commented on core/src/main/resources/hudson/model/AbstractBuild/index.jelly:
```diff
@@ -58,7 +58,7 @@ THE SOFTWARE.
 
         <j:set var="set" value="${it.changeSet}" />
-        <t:summary icon="symbol-list">
+        <t:summary icon="symbol-changes">
```
That doesn't look better either imo, not every change is caused by a PR.
What do you or others think about using `document-text`, `documents` or the already added details one?

janfaracik replied:
Hmm might be a little too similar to the newer 'Status' icon, unless there's a better icon for that too? - 

<img width="340" alt="image" src="https://user-images.githubusercontent.com/43062514/156944230-c2a4fb20-5c03-4ae9-8fe6-fed9e179a63f.png">

NotMyFault replied:
I was referring to this one:
![](https://i.imgur.com/X2mAAfu.png)

Maybe try that one fits all occasions?
![](https://i.imgur.com/hWOC0RG.png)

janfaracik replied:
Looks good to me! Done :)

NotMyFault commented on core/src/main/resources/hudson/model/AbstractBuild/index.jelly:
```diff
@@ -58,7 +58,7 @@ THE SOFTWARE.
 
         <j:set var="set" value="${it.changeSet}" />
-        <t:summary icon="symbol-list">
+        <t:summary icon="symbol-changes">
```
That doesn't look better either imo, not every change is caused by a PR.
What do you or others think about using `document-text` or `documents` ?

NotMyFault replied:
That doesn't look better either imo, not every change is caused by a PR.
What do you or others think about using `document-text` or `documents` ?

NotMyFault commented on core/src/main/resources/hudson/model/DirectoryBrowserSupport/dir.jelly:
```diff
@@ -33,7 +33,7 @@ THE SOFTWARE.
         <div class="parentPath">
           <form action="${backPath}" method="get">
-            <a href="${topPath}"><l:icon class="${icons.toNormalizedIconNameClass(icon)} icon-xlg" /></a>
+            <a href="${topPath}"><l:icon class="${icons.toNormalizedIconNameClass(icon)} icon-md" /></a>
```
Unfortunately, that doesn't make it much better either 
![](https://i.imgur.com/zu8PIzT.png)

janfaracik replied:
I've updated the page, it's not perfect (I want to take another look at this page in the future) but I think it's cleaner now:

<img width="590" alt="image" src="https://user-images.githubusercontent.com/43062514/155698536-45f39264-56ca-4f72-8b1f-288ec1a7d980.png">

NotMyFault replied:
That looks much better <3

janfaracik committed:
Update build history icon, potential fix for credentials oversize

NotMyFault commented on war/src/main/less/modules/side-panel-tasks.less:
```diff
@@ -76,6 +76,8 @@
     margin: 0;
     width: 20px;
     height: 20px;
+    max-width: 20px;
+    max-height: 20px;
```
If we move this to the block below, we _should_ be good to go, based on a quick look.

Guess, we could even go with 24px, to simulate the size of a medium-sized icon.

fqueiruga commented on war/src/main/less/modules/buttons-temp.less:
```diff
@@ -0,0 +1,133 @@
+}
+
+@variants: {
+  destructive: #fc2125;
```
can't this be a CSS var? it's used with an !important so it should be themeable

janfaracik replied:
Ah that code is from a time before the colour vars :P I've updated it - thanks for the heads up :)

fqueiruga commented on war/src/main/resources/images/symbols/rss.svg:
```diff
@@ -0,0 +1,10 @@
+<svg width="512px" height="512px" viewBox="0 0 512 512" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
+        <g transform="translate(81.000000, 81.323000)" fill-rule="nonzero" stroke="currentColor" stroke-width="32">
+            <path d="M28.552,293.458 L28.2086637,293.462282 C12.7932399,293.768174 0.411027957,306.374012 0.439973838,321.849687 C0.469416663,337.481891 13.1509336,350.138264 28.7831021,350.136598 C44.2589396,350.13495 56.8401627,337.727807 57.1154778,322.311817 L57.119,321.927 L57.1149621,321.530547 C56.8237248,306.313721 44.6521378,293.97949 29.47116,293.473221 L29.045,293.462 L28.552,293.458 Z ...
```
This SVG could be optimized, I'd suggest using https://jakearchibald.github.io/svgomg/

janfaracik replied:
I ran it through that service but it seemingly broke the SVG? :P I've updated the icon to look a little nicer.

janfaracik committed:
Update design of workspace

janfaracik committed:
Update buttons-temp.less

timja committed:
Fix tests

timja committed:
Merge branch 'master' into new-task-design

janfaracik committed:
Update core/src/main/resources/hudson/model/Label/sidepanel.jelly

Co-authored-by: Daniel Beck <1831569+daniel-beck@users.noreply.github.com>

janfaracik committed:
Fix oversized credentials icon

janfaracik committed:
Merge branch 'new-task-design' of https://github.com/janfaracik/jenkins into scalable-icon-support-2-sidebar

janfaracik committed:
Set title rather than use variable

janfaracik committed:
Update people.svg

janfaracik committed:
Optimize RSS

janfaracik committed:
Update rss.svg

janfaracik committed:
Shrink RSS icon

janfaracik committed:
Update sidepanel.jelly

janfaracik committed:
Update summary.jelly

janfaracik committed:
Update 'Update Center' icon to be a download icon

janfaracik committed:
Update info icon

janfaracik committed:
Update artifactList.jelly

janfaracik commented:
> Plugin Manager and Update Center have the same icon (a puzzle piece). Is there no "Download" looking icon (e.g. a bold downward arrow outline) available for the latter? (Seems not introduced here.)

Good spot, I've updated it 👍 

<img width="282" alt="image" src="https://user-images.githubusercontent.com/43062514/156649895-9b1e7f50-b9a5-46b3-8438-f750c38abde5.png">

> In case the following are unintentionally unchanged:
> 
> * The "right arrow" icons on the "Update Center" page related to Jenkins restart are unchanged and look out of place.
> * <img alt="Screenshot 2022-02-24 at 20 39 40" width="548" src="https://user-images.githubusercontent.com/1831569/155595628-682789d0-bc0f-4f9f-b077-300abc10b2bb.png">

I've updated this, would you mind taking another look? Not too sure on the icon at the mo.

NotMyFault reviewed (commented):
- The optimization of the rss feed icon somehow did break it:
![](https://i.imgur.com/sYOZrzL.png)
Restoring to 1257aa6bc202e9ff68cce63b2eb4e5c571c3ff4a resolves it.

Also, what do you think about my comment on the changeset icon?
https://github.com/jenkinsci/jenkins/pull/6307#discussion_r814364992

timja commented:
> * The optimization of the rss feed icon somehow did break it:
>   ![](https://camo.githubusercontent.com/6059e60533b3e93cd061eaa4323ef5f66691abf9e567da7e722acf77fed9e4e0/68747470733a2f2f692e696d6775722e636f6d2f73594f5a727a4c2e706e67)
> 
> Also, what do you think about my comment on the changeset icon? [#6307 (comment)](https://github.com/jenkinsci/jenkins/pull/6307#discussion_r814364992)

It works on dark theme:
![image](https://user-images.githubusercontent.com/21194782/156841100-08801ca4-cf56-4d3d-8ea0-79846e659539.png)

Probably why was missed

timja commented:
💩 build failing because of artifactory being down

timja commented:
Looking really good

Only issues I found are I think pre-existing issues from recent UI changes but I'll mention them here in case they are quick fixes:

![image](https://user-images.githubusercontent.com/21194782/156875151-e9bcf861-b1e4-4473-9bec-66061ddaac07.png)

Guessing that 'Move' comes from a plugin (appears inside folders)?
![image](https://user-images.githubusercontent.com/21194782/156875187-580877ac-a60e-46a3-b231-b20c56079e06.png)

Symbols on the run summary page are a bit mis-aligned but pre-existing issues:

![image](https://user-images.githubusercontent.com/21194782/156875218-567223db-9cd1-420a-bac6-3b20f95f5f61.png)

timja committed:
Add support for symbols in Actions

timja commented:
I pushed a commit https://github.com/jenkinsci/jenkins/pull/6307/commits/4b398ee86871e789c23bf3a025f8dfe239659624 to add support for attaching an action, changing this https://github.com/jenkinsci/credentials-plugin/blob/master/src/main/java/com/cloudbees/plugins/credentials/ViewCredentialsAction.java#L113-L117 didn't work otherwise.

NotMyFault commented:
> Only issues I found are I think pre-existing issues from recent UI changes but I'll mention them here in case they are quick fixes:
> 
> ![image](https://user-images.githubusercontent.com/21194782/156875151-e9bcf861-b1e4-4473-9bec-66061ddaac07.png)

I took care of that a bit ago, was released as 2.4.0 of the favorite plugin.

> 
> Guessing that 'Move' comes from a plugin (appears inside folders)? ![image](https://user-images.githubusercontent.com/21194782/156875187-580877ac-a60e-46a3-b231-b20c56079e06.png)

Afaik, folders and credentials has one. But you'll likely find quite a lot of plugins adding own icons to the sidebar, that don't match the newest designs yet.

timja commented:
I’ve got some changes locally for credentials plugin any suggestions for the move icon and the edit credentials button?

NotMyFault commented:
> I’ve got some changes locally for credentials plugin any suggestions for the move icon and the edit credentials button?

Isn't the "edit credentials" the setting icon from core? Once overridden, no more work should be needed I guess?
For the former, the swap or some other arrow icon would be fine.

timja commented:
It is yeah just need one to override it with

NotMyFault commented:
The build one looks fine imo
![](https://i.imgur.com/FkuFoHQ.png)

timja committed:
Add more symbols

timja commented:
Opened https://github.com/jenkinsci/credentials-plugin/pull/287 with what I have so far 😄.

Looks a lot better imo

timja committed:
Merge branch 'master' into new-task-design

timja commented:
Icons are still blue on user dropdown:

<img width="203" alt="image" src="https://user-images.githubusercontent.com/21194782/156899923-2ba85c67-25b1-44fe-b392-491c060db0a5.png">

timja committed:
More icons

timja committed:
Use modern help icon

timja committed:
Make it easier for generic icon loading to work

daniel-beck commented:
Regression:
<img width="587" alt="Screenshot 2022-03-06 at 12 44 03" src="https://user-images.githubusercontent.com/1831569/156921557-da9422fe-c6f3-413d-878c-678f8865a3b6.png">
In 2.337:
<img width="516" alt="Screenshot 2022-03-06 at 12 45 28" src="https://user-images.githubusercontent.com/1831569/156921596-e37091e6-312a-46b6-861f-e915949a9831.png">

Please also figure out a way other than "look at the commit history" to tell plugin maintainers what version of Jenkins introduced an icon that they are able to use.

timja committed:
Apply patch

NotMyFault commented:
I also noticed a lot of tooltips vanishing, especially from the sidebar:

https://user-images.githubusercontent.com/13383509/156941445-ef4c7deb-5514-4345-929b-17302bfad653.mp4

timja commented:
Tbh those tooltips are useless and shouldn't be there in the sidebar anyway. If you find any places where information is being lost that would be useful 👍

NotMyFault commented:
> Tbh those tooltips are useless and shouldn't be there in the sidebar anyway. If you find any places where information is being lost that would be useful 👍

Your credentials PR, the type of credential and the provider information is dropped. 

https://user-images.githubusercontent.com/13383509/156942295-30b0b1f0-f45d-454b-94c4-f2b594111bdb.mp4

janfaracik committed:
Fix RSS color

janfaracik committed:
Add tooltip support for symbols

janfaracik commented:
> > Tbh those tooltips are useless and shouldn't be there in the sidebar anyway. If you find any places where information is being lost that would be useful 👍
> 
> Your credentials PR 👀 The type of credential and the provider information is dropped.
> 
>  2022-03-06.22-06-14.mp4

Fixed in the latest commit - thanks for finding :) 

<img width="249" alt="image" src="https://user-images.githubusercontent.com/43062514/156943566-403ceb91-6bea-4184-9494-d6b464154572.png">

janfaracik committed:
Fix ID Card color

janfaracik committed:
Add unit tests for symbols

timja committed:
Minor refactor, sort translations, add another translation

janfaracik committed:
Merge branch 'master' into scalable-icon-support-2-sidebar

janfaracik commented:
> Icons are still blue on user dropdown:
> 
> <img alt="image" width="203" src="https://user-images.githubusercontent.com/21194782/156899923-2ba85c67-25b1-44fe-b392-491c060db0a5.png">

I'll update that when I get to the replacing YUI menus/tooltips MR if that's okay 🙌

janfaracik committed:
Update changes icon

janfaracik commented:
> Looking really good
> 
> Only issues I found are I think pre-existing issues from recent UI changes but I'll mention them here in case they are quick fixes:
> 
> ![image](https://user-images.githubusercontent.com/21194782/156875151-e9bcf861-b1e4-4473-9bec-66061ddaac07.png)
> 
> Guessing that 'Move' comes from a plugin (appears inside folders)? ![image](https://user-images.githubusercontent.com/21194782/156875187-580877ac-a60e-46a3-b231-b20c56079e06.png)
> 
> Symbols on the run summary page are a bit mis-aligned but pre-existing issues:
> 
> ![image](https://user-images.githubusercontent.com/21194782/156875218-567223db-9cd1-420a-bac6-3b20f95f5f61.png)
> 
> New item inside a folder uses the old icon:
> 
> ![image](https://user-images.githubusercontent.com/21194782/156876102-dc9f3336-ba7e-48ce-bf7b-4361bcb37593.png)

Once this PR is merged I'm hoping to get a discussion/PR up for the custom symbols for plugins so should be good to update those icons then.

timja committed:
Merge branch 'master' into new-task-design

janfaracik committed:
Fix oversized log help button

janfaracik committed:
Update log recorder new button

janfaracik commented on core/src/main/resources/hudson/model/DirectoryBrowserSupport/dir.jelly:
```diff
@@ -87,25 +89,26 @@ THE SOFTWARE.
-                        <a href="${x.href}/*view*/">${%view}</a>
-                      </j:if>                      
+                        <a class="jenkins-link--with-icon" href="${x.href}/*view*/" tooltip="${%View}">
+                          <l:icon src="symbol-download" />
```
Sounds good to me :) Done

<img width="441" alt="image" src="https://user-images.githubusercontent.com/43062514/157315860-88e85b7e-bc83-4f36-a81a-dee0cb5fed2d.png">

janfaracik committed:
Use view icon instead of download for workspace

janfaracik committed:
Update logger configuration help icon

NotMyFault commented on core/src/main/resources/lib/hudson/help.jelly:
```diff
@@ -33,10 +33,10 @@ THE SOFTWARE.
   </st:documentation>
-  <a href="${href}">
-    <l:helpIcon iconSize="${iconSize}" class="icon-help" tooltip="${tooltip}"/>
+  <a href="${attrs.href}" class="jenkins-help-button ${attrs.iconSize}" tooltip="${attrs.tooltip}">
+    ?
```
Looks really nice!
Maybe it could be possible to have that one available as jelly tag or to be the default over the round help icons in the future 👀

janfaracik replied:
Just finished a small follow up branch to do just that - thanks! :)

alecharp replied:
I found a similar issue on the log recorder configuration page: 

![Screenshot from 2022-03-11 11-55-39](https://user-images.githubusercontent.com/985955/157854090-59b5dacf-5bb7-45ae-ace5-71716ee2b388.png)

but I cannot point the issue to a file change in this PR. Maybe it's my eyes, but I see the `?` icon to be different in size from the others I can find. 

I tested on Fedora with Firefox 97 using the `jenkins/core-pr-tester` Docker image.

NotMyFault replied:
> Maybe it's my eyes, but I see the `?` icon to be different in size from the others I can find.

That's caused by the header size, I guess. If I change it from h1 to h4, it displays the typical help button like shown in other occasions:
![](https://i.imgur.com/phSA0Ww.png)

timja replied:
fixed

NotMyFault reviewed (approved):
🎉

timja committed:
Fix spotless

alecharp commented on core/src/main/java/org/jenkins/ui/icon/IconSet.java:
```diff
@@ -535,19 +545,38 @@ private static void initializeSVGs() {
-        if (tangoIcon != null) {
-            tangoIcon = tangoIcon.split(" ")[0];
-        }
+        String cleanedTangoIcon = cleanName(tangoIcon);
```
```suggestion
```

alecharp commented on core/src/main/java/org/jenkins/ui/icon/IconSet.java:
```diff
@@ -535,19 +545,38 @@ private static void initializeSVGs() {
-        translations.put("icon-help", "symbol-help-circle");
+        translations.put("icon-user", "symbol-people");
+
+        return translations.getOrDefault(cleanedTangoIcon, null);
```
```suggestion
        String cleanedTangoIcon = cleanName(tangoIcon);
        return translations.getOrDefault(cleanedTangoIcon, null);
```

alecharp reviewed (changes requested):
I would have been fine with the code as is, but apparently checkstyle is not.

timja committed:
Apply suggestions from code review

Co-authored-by: Adrien Lecharpentier <adrien.lecharpentier@gmail.com>

timja committed:
Fix help icon rendering and cleanup icons for logging

timja committed:
Update missed icon

timja committed:
Use jelly tag for app bar

uhafner reviewed (commented):
I finally found some time to look at those new changes. In general I like the new icons and the new coloring in black. I have some comments (or requests):

- The side panel icons are too small now, just 20x20 (24x24 before). It is hard to identify some icons due to this scaling. Especially the ionic icons are so thin that they do not look good on such scaling anymore (FontAwesome icons look much better on this scaling :wink:). 
- Interestingly, the side panel links require more vertical space now, even though the icons are smaller? I think a lot of people already complained about the padding, this makes it even worse.
- The side panel is missing the vertical line now, we discussed that topic in the UX meetings quite some time ago and came to the conclusion that a vertical line helps to improve the usability.
- The artifacts summary now uses a link for the icon, which makes the icon show in blue. This should be reverted.

I wonder why this PR also changes the items on the right side (main view)? This part seems to be totally broken now. The original icons use a size of 48x48 but now the new icons use 32x32 and are not centered. This looks totally weird if mixed with existing plugins. I think we should define first how we we would like to use `summary.jelly` files in the future before we change the format of those boxes! Maybe these boxes should be some kind of cards (bootstrap) with a title, icon and content. 

![Bildschirmfoto 2022-03-14 um 10 03 37](https://user-images.githubusercontent.com/503338/158139380-392e25c8-f830-4e7d-b262-4e2984b1337d.png)

janfaracik committed:
Increase icon size a touch, now use REM for icon sizing, decrease padding slightly

timja committed:
Fix icon size, remove script console icon

uhafner commented:
I also think that using black as icon color is a little bit to harsh (I'm not sure if that is the right English word). Gray might look better (for the summary icons). It would be also helpful if the `t:summary` tag would be changed so that plugin SVG icons will use the same styling, see https://issues.jenkins.io/browse/JENKINS-67844.

![Bildschirmfoto 2022-03-15 um 21 18 37](https://user-images.githubusercontent.com/503338/158464562-25960ebc-721a-40d9-9f6d-b086ec2a4b42.png)

janfaracik committed:
Fix summary icon sizes

timja commented on war/src/main/less/base/style.less:
```diff
@@ -1659,4 +1659,15 @@ svg.icon-xlg {
   display: none;
 }
 
+.app-summary {
+  & > td:first-of-type {
+    img,
+    svg {
+      width: 48px !important;
+      height: 48px !important;
+      margin-right: 1rem;
+    }
+  }
+}
```
is this actually needed? I updated the icon class to use xlg in https://github.com/jenkinsci/jenkins/pull/6307/commits/35510e0e45f53fb0e7dfb41018339007e275e60a which fixed the issue afaik

janfaracik replied:
From what I saw it was causing a misalignment issue on pipeline jobs.

janfaracik committed:
Update started by icon

janfaracik committed:
Update project relationship icon

janfaracik commented:
> Some comments about some icons:
> 
> * The play icon is currently used to start a build. It should not be used to show the trigger of a build, this is misleading.

I've updated that now, it looks like so - 

<img width="435" alt="image" src="https://user-images.githubusercontent.com/43062514/158477386-b8d6b21f-a933-4443-ae92-32c19ae0b969.png">

> * The search icon is used for the _Status_ and the _Project Relationship_ side panel link of a pipeline, this is misleading.

Updated this for project relationship - 

<img width="227" alt="image" src="https://user-images.githubusercontent.com/43062514/158477467-27bf3eb9-94e0-4472-bde2-b0ac4f88e0cc.png">

Updating the status icon will require a plugin update sadly as they're referencing `icon-search`, so I can't replace it on Jenkins side without breaking all the other search icons.

> * _Pipeline Steps_ in the side panel is using the configure icon but is is a read only view, this is misleading.

Haven't looked too deep into this one but I imagine it's the same as the `icon-search` issue above.

> * Workspace and recent changes: the icon and the text is clickable (and blue). Only the text should be clickable.

That's how it works by default (unless I changed it and don't remember) https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/lib/hudson/summary.jelly

Happy to remove the links from the icons, I wouldn't mind taking another look at summary.jelly in the future though as its still using `tables` etc.

> * In pipelines there is a changes link which still uses the tango icon, but recent changes correctly uses a new icon.

Same thing as the search icon, pipeline references icon-notepad.

> * _New View_ icon should have a plus sign in it. (Actually the side panel link is obsolete and should be removed)

I've got a tiny PR in the works that actually addresses these sorts of things across Jenkins, essentially removing links that are duplicated on the page just like the New View one.

timja commented:
> I wouldn't mind taking another look at summary.jelly in the future though as its still using `tables` etc.

I have a PR here from awhile back that fixes that could revive it at some point:
https://github.com/jenkinsci/jenkins/pull/5351

timja committed:
Merge branch 'master' into new-task-design

timja committed:
Update 'View as plain text'

timja committed:
Merge branch 'new-task-design' of github.com:janfaracik/jenkins into new-task-design

timja committed:
Remove link from summary

timja reviewed (approved):
'Orange square' isn't currently replaced.

Not really sure what it's supposed to mean, seems to be used when someone aborts a build at least:

<img width="262" alt="image" src="https://user-images.githubusercontent.com/21194782/158484303-a6ef284a-feeb-434e-8a2f-cd12eaf7580f.png">

----

I've gone through a lot of pages with quite a few plugins installed. Looks really good imo.
We can fix other issues in follow-ups unless anyone has any blockers.

Let's show it at the ux-sig tomorrow and see what people think anyway

NotMyFault commented:
> 'Orange square' isn't currently replaced.
> 
> Not really sure what it's supposed to mean, seems to be used when someone aborts a build at least:
> 

It's also used if the build wasn't triggered by a user but an SCM change or push event e.g.
However, using a crosshair here is a bit weird, I guess. Though arguably an orange square isn't much specific either...

There are a few occasions where I'd vouch for different icon choices used in the main panel, yet that is no merge blocker and only my two cents, which always could be fine-tuned later one.

timja commented:
@uhafner are any of your comments blockers or do you think it's in a state that we could ship and iterate?

----

> I also think that using black as icon color is a little bit to harsh

For me I find it okay, it looks a bit weird when mixed amongst non adapted icons which will get sorted after merge as plugins are adapted. 

Grey does look okay though, I tried it with `#737373`

![image](https://user-images.githubusercontent.com/21194782/158542248-ab4f7fff-5bea-4713-8484-5b619f808f5c.png)

![image](https://user-images.githubusercontent.com/21194782/158542361-4f0655ec-2bc9-4ff8-8508-1c8a5ac7f164.png)


> (I'm not sure if that is the right English word).
Yes the word is fine

uhafner commented:
> I think unless the icons are of type svgIcon or a Jenkins Symbol they won't have the colouring applied sadly, this shouldn't be an issue once custom symbol support is added to Jenkins (hopefully fairly soon after this PR is merged).

I think a simple `if (icon.hasSuffix("svg")) then renderAsSVG()` should do the trick in core. But anyway, I can wait for the custom icons PR 👍

uhafner reviewed (approved):
> @uhafner are any of your comments blockers or do you think it's in a state that we could ship and iterate?@uhafner are any of your comments blockers or do you think it's in a state that we could ship and iterate?

No, I'm fine with this PR now!

janfaracik committed:
Merge branch 'master' into scalable-icon-support-2-sidebar

timja commented:
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

----

@daniel-beck feel free to remove if you want to still review, but we can always address afterwards / in the followup PR

There's a branch here that allows plugins to add their own symbols: https://github.com/janfaracik/jenkins/tree/scalable-icon-support-3-custom-symbols
https://github.com/jenkinsci/ui-samples-plugin/pull/14 is using that branch.

basil commented:
Lacked sufficient testing. Causes new PCT failures in:

- `com.cloudbees.jenkins.support.actions.SupportAbstractItemActionTest#onlyAdminCanSeeAction`
- `com.cloudbees.jenkins.support.actions.SupportComputerActionTest#onlyAdminCanSeeAction`
- `com.cloudbees.jenkins.support.actions.SupportRunActionTest#onlyAdminCanSeeAction`

See [PR build 1](https://ci.jenkins.io/job/Tools/job/bom/job/PR-955/1/) of jenkinsci/bom#955.

alecharp commented:
@janfaracik just wondering why you added that commit here? The PR was merged. Is there something I'm missing?

janfaracik commented:
> @janfaracik just wondering why you added that commit here? The PR was merged. Is there something I'm missing?

Ah think I had done a dodgy merge in another branch that referenced this.

ViliusS commented:
I totally don't understand why these new icons are introduced. Loss of color makes them difficult to read and almost inseparable one from the other. Some icons are actually the same for different actions (like People and My Views menu item, or Configuration and View Configuration). Some doesn't match action at all (like build Status icon on the left menu which is now represented by search hourglass).

daniel-beck commented on core/src/main/java/org/jenkins/ui/icon/IconSet.java:
```diff
@@ -77,16 +77,22 @@ private static String prependTitleIfRequired(String icon, String title) {
             symbol = symbol.replaceAll("(class=\")[^&]*?(\")", "$1$2");
+            symbol = symbol.replaceAll("(tooltip=\")[^&]*?(\")", "");
+            if (!tooltip.isEmpty()) {
+                symbol = symbol.replaceAll("<svg", "<svg tooltip=\"" + tooltip + "\"");
```
Caused [SECURITY-2776](https://www.jenkins.io/security/advisory/2022-06-22/#SECURITY-2781)

daniel-beck commented:
Caused [JENKINS-70191](https://issues.jenkins.io/browse/JENKINS-70191).

daniel-beck commented:
Caused [JENKINS-70246](https://issues.jenkins.io/browse/JENKINS-70246).

daniel-beck commented on core/src/main/resources/lib/hudson/summary.jelly:
```diff
@@ -44,20 +44,9 @@ THE SOFTWARE.
-                      alt="" style="width: 48px; height: 48px; margin-right:1em;" />
-            </j:otherwise>
-          </j:choose>
+          <l:icon src="${icon}" />
```
Caused [JENKINS-75603](https://issues.jenkins.io/browse/JENKINS-75603).
